src/ride/track_paint.c

```diff
@@ -1139,3 +1144,321 @@ TRACK_PAINT_FUNCTION get_track_paint_function_facility(int trackType, int direct
 	}
 	return NULL;
 }
+
+/* rct2: 0x00518394 */
+void junior_rc_flat_paint_setup(uint8 rideIndex, uint8 trackSequence, uint8 direction, int height, rct_map_element* mapElement){
+
+	uint32 image_id = SPR_JUNIOR_RC_FLAT_NE_SW | RCT2_GLOBAL(0x00F44198, uint32);
+	image_id += direction & 1;
+
+	if (mapElement->type & (1 << 7)) {
+		image_id += 106;
+	}
+
+	//di
+	uint8 lengthX = direction & 1 ? 20 : 32;
+	//si
+	uint8 lengthY = direction & 1 ? 32 : 20;
+
+	//al
+	sint8 offsetX = direction & 1 ? 6 : 0;
+	//cl
+	sint8 offsetY = direction & 1 ? 0 : 6;
+
+	sub_98196C(image_id, offsetX, offsetY, lengthX, lengthY, 1, height, get_current_rotation());
+
+	if ((RCT2_GLOBAL(0x009DE56A, sint16)& (1 << 5) && !(RCT2_GLOBAL(0x009DE56E, sint16)& (1 << 5))) || 
+		!(RCT2_GLOBAL(0x009DE56A, sint16)& (1 << 5) && (RCT2_GLOBAL(0x009DE56E, sint16)& (1 << 5)))) {
+		int edi = direction & 1 ? 2 : 1;
+		RCT2_CALLPROC_X(0x00663105, 0, 4, 0, height, 20, edi, RCT2_GLOBAL(0x00F4419C, uint32));
```

Comment:
This should be `metal_a_wooden_a_supports_paint_setup`. `ebx` is the desired segment to put the pole on. 
```
    0
  5   6
1   4   2
  7   8
    3
```